scripts/20170928-col1108/clean_up_canvas_after_s3_migration.js

```diff
@@ -0,0 +1,434 @@
+ *
+ * @param  {Course}           course              The course to update
+ * @param  {Number}           index               Index in the list of courses, used to show progress
+ * @param  {Function}         callback            Standard callback function
+ * @return {Object}                               Return per callback
+ */
+var deleteCanvasFolderPerCourse = function(course, index, callback) {
+  log.info({
+    'course': course.id,
+    'canvas_course_id': course.canvas_course_id,
+    'canvas_api_domain': course.canvas_api_domain
+  }, 'Prepare to clean up course \'' + course.name + '\'');
+
+  var folderName = '_suitec';
+  var getFolderUrl = util.format('%s/api/v1/courses/%d/folders/by_path/%s',
+    getCanvasBaseURI(course.canvas),
+    course.canvas_course_id,
+    folderName);
+  var getFolderOpts = {
+    'url': getFolderUrl,
+    'method': 'GET',
+    'headers': {
+      'Authorization': util.format('Bearer %s', course.canvas.api_key)
+    }
+  };
+
+  request(getFolderOpts, function(err, response, body) {
+    if (err) {
+      log.error({
+        'err': err,
+        'course': course.id,
+        'canvas_course_id': course.canvas_course_id,
+        'folderName': folderName
+      }, 'Failed to check existence of Canvas folder');
+
+      recordFailure(course, 'Err: ' + err.message);
+
+      return callback();
+
+    } else if (response.statusCode === 404) {
+      log.info({
+        'course': course.id,
+        'canvas_course_id': course.canvas_course_id,
+        'folderName': folderName
+      }, 'Canvas folder not found for course');
+
+      notFound.push([course.id, course.canvas_course_id, course.name]);
+
+      return callback();
+
+    } else if (response.statusCode !== 200) {
+      log.error({
+        'response': response,
+        'course': course.id,
+        'canvas_course_id': course.canvas_course_id,
+        'folderName': folderName
+      }, 'Failed to check existence of Canvas folder; unexpected status code');
+
+      recordFailure(course, 'Unexpected status code: ' + response.statusCode);
+
+      return callback();
+    }
+
+    var folderInfo = null;
+    try {
+      folderInfo = JSON.parse(body);
+    } catch (parseErr) {
+      log.error({'err': parseErr, 'course': course.id}, 'Failed to parse Canvas response');
+
+      recordFailure(course, 'Failed to parse Canvas response. Parse err: ' + parseErr.message);
+
+      return callback();
+    }
+
+    var folderId = _.last(folderInfo).id;
+
+    emphatic(util.format('Course \'%s\' (%d, %d) has folder: %s',
+      course.name,
+      course.id,
+      course.canvas_course_id,
+      folderName));
+
+    var deleteFolderUrl = util.format('%s/api/v1/folders/%d', getCanvasBaseURI(course.canvas), folderId);
+    var opts = {
+      'url': deleteFolderUrl,
+      'method': 'DELETE',
+      'headers': {
+        'Authorization': util.format('Bearer %s', course.canvas.api_key)
+      },
+      'qs': {
+        'force': 'true'
```

Comment:
If `_suitec` folder is non-empty then Canvas requires the `force` arg.  I verified on my local Canvas instance.